### PR TITLE
rootless.md: massive update

### PR DESCRIPTION
**Pull request by AkihiroSuda**
### Proposed changes

* Add Ubuntu 20.04
* Remove Fedora 30 (EOL)
* `docker run --net=host` does not work in the most expected way
* Allow installation as the root
* Nightly channel
* `docker context create rootless`
* `DOCKER_HOST=ssh://...`
* Alternatives to cgroup flags (`docker run --cpus --memory --pids-limit`)
* A bunch of troubleshooting tips

### Related issues (optional)

Fix https://github.com/docker/docker.github.io/issues/10840

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
@thaJeztah @tonistiigi @tiborvass PTAL

**Comment by AkihiroSuda** [comment]
Readability might not be good :(
https://deploy-preview-10965--docsdocker.netlify.app/engine/security/rootless/#troubleshooting

**Comment by AkihiroSuda** [comment]
@thaJeztah @tonistiigi @tiborvass PTAL?

**Comment by thaJeztah** [comment]
oh! I had some comments, but see I didn't submit them; doing so now

File: engine/security/rootless.md [hunk 1]
```diff
@@ -258,11 +300,23 @@ Or add `net.ipv4.ip_unprivileged_port_start=0` to `/etc/sysctl.conf` (or
...
-can be still used, though they work in process-granularity rather than in container-granularity.
+can be still used, though they work in process-granularity rather than in container-granularity,
+and can be arbitrary disabled by the container process.
+
+e.g.
+- To limit CPU usage to 0.5 cores (akin to `docker run --cpus 0.5):
+  `docker run <IMAGE> cpulimit --limit=50 --include-children <COMMAND>`
+
+- To limit max VSZ to 64MiB (akin to `docker run --memory 64m`):
+  `docker run <IMAGE> sh -c "ulimit -v 65536; <COMMAND>"`
```
[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply engine/security/rootless.md:315]: No: https://github.com/docker/go-units/blob/519db1ee28dcc9fd2474ae59fca29a810482bfb1/ulimit.go#L47

File: engine/security/rootless.md [hunk 2]
```diff
@@ -112,9 +111,12 @@ testuser::231072:65536
 
 - Only `vfs` graphdriver is supported. However, on Ubuntu and Debian 10,
   `overlay2` and `overlay` are also supported.
+   Docker 20.0X will additionally support `fuse-overlayfs` driver to provide
+   overlayfs functionality for any distribution with kernel 4.18 or later.
```
[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply engine/security/rootless.md]: Removed 20.0X parts. (backed up as https://github.com/AkihiroSuda/docker.github.io/commit/e228e903c85d76b53d99266c16ad0681d617fc77)

I'll open 20.0X draft PR after this PR gets merged.

File: engine/security/rootless.md [hunk 3]
```diff
@@ -124,6 +126,10 @@ testuser::231072:65536
 
 - To expose privileged TCP/UDP ports (< 1024), see [Exposing privileged ports](#exposing-privileged-ports)
 
+- `IPAddress` shown in `docker inspect` is namespaced inside RootlessKit's network namespace.
+
+- Host network (`docker run --net=host`) is namespaced inside RootlessKit and works in mostly unexpected way.
```
[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply engine/security/rootless.md]: updated

File: engine/security/rootless.md [hunk 4]
```diff
@@ -208,7 +231,15 @@ You need to set the socket path explicitly.
...
-$ docker run -d nginx
+$ docker run -d -p 8080:80 nginx
+```
+
+To use `docker context` with the rootless socket:
+```console
+$ docker context create rootless --description "for rootless mode" --docker "host=unix://$XDG_RUNTIME_DIR/docker.sock"
+rootless
+Successfully created context "rootless"
+$ docker --context=rootless run -d -p 8080:80 nginx
```
[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply engine/security/rootless.md]: updated

File: engine/security/rootless.md [hunk 5]
```diff
@@ -124,6 +126,10 @@ testuser::231072:65536
 
 - To expose privileged TCP/UDP ports (< 1024), see [Exposing privileged ports](#exposing-privileged-ports)
 
+- `IPAddress` shown in `docker inspect` is namespaced inside RootlessKit's network namespace.
+
```
[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply engine/security/rootless.md]: updated

File: engine/security/rootless.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (thaJeztah) [reply engine/security/rootless.md]: Thanks! Sorry for that, I know it's taking longer than I wished for 20.x

File: engine/security/rootless.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (thaJeztah) [reply engine/security/rootless.md:311]: oh, interesting; is that something that can be solved?

**Comment by thaJeztah** [comment]
@usha-mandya PTAL

**Pull request merged**